src/services/item/plugins/published/service.ts

```diff
@@ -34,7 +34,10 @@ export class ItemPublishedService {
       .map(({ member }) => member);
 
     const link = buildPublishedItemLink(item);
-
+    if (!link) {
+      console.error('cannot send published email with undefined link');
+      return;
+    }
```

Comment:
Not sure if this should run only when the function is called. It would make more sense to extract the computation above, and throw directly at the top-level, i.e. when the module is loaded at server start (fail fast).